### PR TITLE
refactor map.jinja and improve packaging

### PR DESCRIPTION
**Pull request by aboe76**
refactored map.jinja to single instance and separated the package install to each state:
- systemd
- systemd:networkd
- systemd:resolved
- systemd:timesyncd

added FORMULAS file.

### CODE REVIEW

File: systemd/osfingermap.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# vim: ft=yaml
+#
+# Setup variables using grains['osfinger'] based logic.
+# You just need to add the key:values for an `osfinger` that differ
+# from `defaults.yaml` + `os_family.yaml` + `osmap.yaml`.
+# Only add an `osfinger` which is/will be supported by the formula
+# (empty `osfinger`s do not need to be listed, just added here as an example).
+---
+Ubuntu-18.04: {}
+CentOS-7: {}
```
Comment (myii) [inline systemd/osfingermap.yaml]: Some formulas use `osfinger` mappings and others use `oscodename` mappings.  I can see advantages to both.  The latter is particularly useful when pulling packages from repos, such as in the [`postgres-formula`](https://github.com/saltstack-formulas/postgres-formula/blob/master/postgres/codenamemap.yaml).  Can't think of a way of using both.  What should be the base recommendation for formulas?

**Comment by myii** [comment]
@aboe76 Eyeball check, everything looks good.  Soon, I will try to pull the changes and check a bit more extensively.

File: systemd/osfingermap.yaml (hunk 1 above)
Comment (myii) [reply systemd/osfingermap.yaml]: Perhaps both types should be left out of formulas until they are actually needed?  Then make the selection at the time of need?  This is a question more than a suggestion.

Comment (aboe76) [reply systemd/osfingermap.yaml]: I just added them for reference, so if one wants to contribute they can see where it could go.

Comment (myii) [reply systemd/osfingermap.yaml]: @aboe76 Yes, that's fine.  But long-term, which is more advantageous -- `osfinger` or `oscodename`?  Can't use both together, right?

**Comment by myii** [comment]
@aboe76 Will try to conduct some further tests now.

**Comment by myii** [comment]
@aboe76 Here's a diff of `map.jinja` before and after:

```diff
--- current
+++ PR
@@ -1,6 +1,7 @@
 {
   "networkd": {
     "path": "/etc/systemd/network",
+    "pkg": {},
     "profiles": {
       "link": {
         "10-dmz": [
@@ -90,7 +91,6 @@
   },
   "pkg": "systemd",
   "pkgs_extra": [
-    "libnss-resolve",
     "python-systemd",
     "python3-systemd"
   ],
@@ -98,8 +98,9 @@
     "libsystemd0"
   ],
   "resolved": {
+    "pkg": "libnss-resolve",
     "resolv_file": "/etc/resolv.conf",
-    "resolv_target": "/run/systemd/resolve/resolv.conf"
+    "resolv_target": "/usr/lib/systemd/resolv.conf"
   },
   "service": {
     "syncthing-someuser": {
@@ -119,6 +120,7 @@
     }
   },
   "timesyncd": {
+    "pkg": {},
     "timezone": "UTC"
   },
   "tofs": {
```

If those are all of the changes that you intended then we're good to go.

**Comment by aboe76** [comment]
@myii thanks for the diff, that help me find the one thing that was missing.

**Comment by myii** [comment]
@aboe76 Your new diff:

```diff
--- current
+++ PR
@@ -1,6 +1,7 @@
 {
   "networkd": {
     "path": "/etc/systemd/network",
+    "pkg": {},
     "profiles": {
       "link": {
         "10-dmz": [
@@ -90,7 +91,6 @@
   },
   "pkg": "systemd",
   "pkgs_extra": [
-    "libnss-resolve",
     "python-systemd",
     "python3-systemd"
   ],
@@ -98,6 +98,7 @@
     "libsystemd0"
   ],
   "resolved": {
+    "pkg": "libnss-resolve",
     "resolv_file": "/etc/resolv.conf",
     "resolv_target": "/run/systemd/resolve/resolv.conf"
   },
@@ -119,6 +120,7 @@
     }
   },
   "timesyncd": {
+    "pkg": {},
     "timezone": "UTC"
   },
   "tofs": {
@@ -145,4 +147,16 @@
       "os_familly"
     ]
   }
+}
+{
+  "path": "/etc/systemd/network",
+  "service": "systemd-networkd",
+  "wait_online": true
+}
+{
+  "resolv_file": "/etc/resolv.conf",
+  "resolv_target": "/run/systemd/resolve/resolv.conf"
+}
+{
+  "timezone": "UTC"
 }
```

Let me know if that's OK now.

**Comment by aboe76** [comment]
@myii yeah, but the strange thing is the test fails on something I can't explain...

**Comment by myii** [comment]
@aboe76 Might be easier to discuss on Matrix.  One change is that these are now `2019.2` minions -- `kitchen-docker` started pulling those a few hours back and `debian-8` is now failing on the `template-formula`, because it can't install `salt-minion`.

**Comment by myii** [comment]
@aboe76 Thanks.

**Pull request merged**